### PR TITLE
chore: Standardize CI with other projects

### PR DESCRIPTION
**Pull request by ghislainbourgeois**

### CODE REVIEW

**Comment by gab-arrobo** [comment]
@ghislainbourgeois, should we close this PR? or should we merge it?

**Comment by ghislainbourgeois** [comment]
It might be easier to still merge this one to finish the work of moving the smaller libs to util, and move to the newer mongo at a slower pace. Unless it is quick and easy to change to the newer lib, I can check with Gulsum tomorrow.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3.7.0
```
Comment (gab-arrobo) [inline .github/workflows/main.yml]:
```suggestion
        uses: golangci/golangci-lint-action@v4.0.0
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  fossa-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: FOSSA scan
+        uses: fossa-contrib/fossa-action@v3
+        with:
+          fossa-api-key: 9dc8fa92e3dd565687317beb87b56d89
```
Comment (gab-arrobo) [inline .github/workflows/main.yml]:
```suggestion
          fossa-api-key: 0c3bbcdf20e157bbd487dae173751b28
```

**Review by gab-arrobo — APPROVED** [review]
+1

**Comment by gab-arrobo** [comment]
@ghislainbourgeois, the license-check action fails because several files do not have copyright and license headers.

**Pull request merged**